### ISSUE TITLE
Viper

### ISSUE THREAD

**Issue by HellDragon-HK** [GreatEmerald/UT3Vehicles#16]
Issues and notes on the Viper from in-game testing

UT2004 Draw  - NA
UT3    Draw  - 1.0?
UT3Port Draw - 1.0 / 1.0 looks right but it might need to be scaled down actually but I don't think so

Good collision it seems but gets damaged easily and quickly by climbing hills (even the small ones leading to nodes on Torlan)
Replaces Axon Manta

  - [Priority]  
  - [Port jumps excessively high, you can almost get on top of the tower in the bases on Torlan in UT2004, in UT3 it jumps just high enough to go over a node, also it can stay airborne for a good length of time in UT2004 while in UT3 it falls fairly quick]  
  - [Driver doesn't really go in the seat very well and needs a custom driving animation, it's got to be possible because look at the hoverboard, it's using a slightly custom animation it seems]
  - [Secondary]  
  - [On UT3 it takes about 3 seconds to turn half way around, on UT2004 1 second or less]  
  - [A lot of team lights on it in UT3 all of which are missing on the port]  
  - [Seems really fast in UT2004 like the noderunner, I don't think the one in UT3 is that fast but could be mistaken]  
  - [Extra]  
  - [I don't get fall damage on UT3 from ejecting but I do on UT2004]

**Comment by Unre-Alex** [comment]
When GPing released beta 1 of the viper it had the right look of projectiles and they could do damage, ever after he applied manta's looking like projectiles they seem to stopped working. I still have that file if you want to check it out.

**Comment by HellDragon-HK** [comment]
Sure I guess, I started a project to save as many ut mods as possible to my public G-drive account so always looking for more

**Comment by HellDragon-HK** [comment]
Eh should have guessed, I just thought since it's my report and I'm doing all the testing right now it was directed to me  (no GE not trying to steal your thunder your're the brains and I'm just a dedicated tester)

**Comment by Unre-Alex** [comment]
Sorry I just signed in and don't now how everything works here yet. [Here's the link for viperB1](http://www.mediafire.com/file/z82s1calwql88bh/UT3ViperB1.zip), i hope GPing won't mind.

**Comment by HellDragon-HK** [comment]
New here as well only been 4 days now I guess and thanks and to GPing if you say no I'll get rid of it

**Comment by GreatEmerald** [comment]
@100GPing100 What's the story behind that? Did you lose the code that @Unre-Alex managed to save or something? :)

@HellDragon-HK About the custom animation, I think the Hoverboard uses one of the other stock animations. I don't think there is one for sitting on a bike, but perhaps there is something more appropriate than the default sitting animation...

And the eject effect might need some work, yes. Same as Scorpion's. Its idea is to make sure you don't get hit by your own explosion. The way it works IIRC is that I give the player an invisible shieldbelt that works against the explosion damage type for a limited amount of time, but the eject also gives extra height which results in taking fall damage... Whether or not it should be taken is another question...

**Comment by HellDragon-HK** [comment]
At some point I think I had a vehicle that had a bike driving animation, I'll keep an eye out for it if it would help.
Well I can find an even higher cliff in UT3 and jump off to see what happens I guess if you need

**Comment by GreatEmerald** [comment]
Yes, both of those would be useful.

**Comment by HellDragon-HK** [comment]
The jumping off you mean heh ok I'll see what I can find, do I need to use both the Scorpion and Viper if possible or would one be enough?

**Comment by GreatEmerald** [comment]
Ideally both.

**Comment by HellDragon-HK** [comment]
Ok using the Scorpion eject I jumped out at the top of heading into the middle mine area of Dusk with 30 armor, after hitting the ground near the edge of South Mine node I have 20 armor and 90 health

**Comment by GreatEmerald** [comment]
Hm, OK, that means that UT3 doesn't save you from fall damage, it's just more lenient with it. So the solution is probably to make the eject in the mod fling you upwards less so you don't normally take landing damage.

**Comment by HellDragon-HK** [comment]
Uh that wouldn't be very high, wouldn't having UT3 style fall damage mutator added be a better idea?

**Comment by GreatEmerald** [comment]
Not really, that would involve modifications to all XPawns, and that's hard. It could be implemented in the future but not as part of the vehicles package, that's way out of scope for it.

**Comment by HellDragon-HK** [comment]
Want me to add it as a to do issue in the weapons package then or something?

**Comment by 100GPing100** [comment]
@GreatEmerald @Unre-Alex The projectile change had to do with bouncing.

In `beta 1` the projectile was a subclass of `FlakChunk` (named `UT3ViperBold`); this had some issues tho, one of them I think was the projectile getting stuck sometimes, another thing was probably it's behaviour under water (behaving like flak chunks).

In `beta 2` what I did was subclass `ONSPlasmaProjectile` and wrote the code for bouncing. As for damage, the code is there, but only for when bouncing, if it has reached the max amount of bounces it explodes, all good, thing is I forgot to add in code that handles hitting a `Pawn`.

In `beta 3` nothing related to the projectile was changed.

One of the `todo` for a next release was fixing up the projectile (including visuals), but there was no following release.

@HellDragon-HK As for downloads links, feel free to distribute them; all that's required is for the original archive to be kept unchanged

**Comment by GreatEmerald** [comment]
@HellDragon-HK Maybe. It's going to be a very low-priority wishlist item, though, because that implies porting over all the other Pawn physics, potentially creating an entirely new package for it.
And you can test the Viper too just to be sure.

@100GPing100 Ah, I see. What about the visuals? Were they there in beta 1?

**Comment by HellDragon-HK** [comment]
Alright I'll go find a Viper and a good cliff then

**Comment by 100GPing100** [comment]
@GreatEmerald Yes, it used the same mesh as the projectile in UT3; I think the mesh is also in the other betas but is just not being used.

**Comment by GreatEmerald** [comment]
@HellDragon-HK Huh. That's odd. I might need to compare source codes, then. Is the height at which you get ejected the same betweent he Scorpion and the Viper?

@100GPing100 Ah, OK, good to know! I wonder if it's here in the repository as well, hmm...

**Comment by HellDragon-HK** [comment]
I'll need to retest to be sure, could be Viper was a little higher since it actually jumps, I'll give it another go and let you know

Any chance it's in the archive you gave since I've got the Nemesis files if you know the name I can look for it

**Comment by 100GPing100** [comment]
@GreatEmerald That's a good question; it's highly likely that I may have based the ejection on the one from the scorpion, thus they may be exactly the same; this in terms of values used, because as mentioned by @HellDragon-HK we get ejected in mid air from the viper, thus in the end the total height is most likely different.

**Comment by HellDragon-HK** [comment]
And this time I used the Viper's eject and have 31 left and I'm fairly sure I was in the same spot go figure :S

**Comment by HellDragon-HK** [comment]
Did something change with the Viper in the last git pull guys because the self destruct mode isn't working now, I hear the beep but see no indication to use it and pressing the normal button just jumps out in mid air and leaves the Viper floating in the air for a long time before it comes back down (this should be used on the Raptor and Cicada actually to give them that soft landing ability), I've tried multiple times and on different sessions,

**Comment by HellDragon-HK** [comment]
Added link for drawscale ukx edit

**Comment by 100GPing100** [comment]
Will just leave this here for reference:
[ ] - 100GPing100/UT3Vehicles#10 Viper - Glide and leave
[ ] - 100GPing100/UT3Vehicles#9 Viper - Nerf glide
[ ] - 100GPing100/UT3Vehicles#8 Viper - Self-destruct roll
[ ] - 100GPing100/UT3Vehicles#7 Viper - Projectile look
[ ] - 100GPing100/UT3Vehicles#6 Viper - Projectile damage
[ ] - 100GPing100/UT3Vehicles#5 Viper - Jump on water
[ ] - 100GPing100/UT3Vehicles#4 Viper - Jumping height
[ ] - 100GPing100/UT3Vehicles#3 Viper - Missing eject HUD
[ ] - 100GPing100/UT3Vehicles#2 Viper - Self-destruct boost direction
[x] - 100GPing100/UT3Vehicles#1 Viper - Nerf turn speed

**Comment by HellDragon-HK** [comment]
It can jump on water but the Manta can't, how did that happen?  Are you positive they can't jump on water, not a whole lot of water in UT3 to test that with and I'm drawing a blank on it, VCTF-Corruption has water but don't think it's deep enough.

**Comment by 100GPing100** [comment]
Yeah, neither the manta nor the viper can jump on water in UT3. You can test it on `VCTF-Suspension`.

**Comment by Unre-Alex** [comment]
@100GPing100 Nice to see you working on the Viper. One more thing that hasn't been mentioned. If you're keep holding a forward key before and after entering the Viper, the "SpeedUp" animation doesn't activate until you press some other movement key or release the forward key and press it again. Also, in UT3 the animation starts instantly while in UT2004 it takes a moment to start.

**Comment by HellDragon-HK** [comment]
And another message I didn't get an e-mail about, here I've been waiting for a reply and it's already here for as long as I asked......ugh

Ok then well I started working on it today and I've got decent progress, will probably have a request up tonight for it....something I've noticed is during jumping the Viper has an animation where it kind of moves downward but the driver doesn't unfortunately and is floating above the Viper for the whole time it's in the air, is that even possible to fix? 

Btw also I've uploaded the movement video, at the bottom of the task list like the others.

**Comment by HellDragon-HK** [comment]
The Viper is missing a single large engine flame below it, I got the Manta & Goliath's engine flames but those are both double, and the only thing I can think of that has a single engine flame are the space fighters, not sure that those will work though.

Any thoughts anyone?
